script/validate_json.rb

```diff
@@ -1,53 +1,79 @@
 #!/usr/bin/env ruby
 
 # Validates JSON files in the _data directory
-# Exits 0 on success
-# Exits 1 upon JSON parsing errors
-# Exits 2 if a file's keys are not in alphanumeric order
-# Exits 3 if any sites.json entries are missing the required 'url' key
 
 require 'json'
 
+module ExitCodes
+    SUCCESS = 0
+    PARSE_FAILED = 1           # JSON parse errors
+    UNSORTED = 2               # data keys are not in alphanumeric order
+    MISSING_URL = 3            # Entry missing the required 'url' field
+    MISSING_DIFFICULTY = 4     # Entry missing the required 'difficulty' field
+    MISSING_DOMAINS = 5        # Entry missing the required 'domains' field
+    MISSING_NAME = 6           # Entry missing the required 'name' field
+    UNEXPECTED_DIFFICULTY = 7  # Unexpected value for 'difficulty' field
+end
+
+SupportedDifficulties = ["easy","medium","hard","impossible"]
+
 def get_transformed_name(site_object)
     return site_object['name'].downcase.sub(/^the\s+/, '')
 end
 
+def error_on_missing_field(key, field, exit_code)
+    unless key.key?(field)
+        STDERR.puts "Entry '#{key['name']}' has no '#{field}' field"
+        exit exit_code
+    end
+end
+
+def validate_website_entry(key, i)
+    unless key.key?('name')
+        STDERR.puts "Entry #{i} has no 'name' field"
+        exit ExitCodes::MISSING_NAME
+    end
+    error_on_missing_field(key, 'url', ExitCodes::MISSING_URL)
+    error_on_missing_field(key, 'difficulty', ExitCodes::MISSING_DIFFICULTY)
+    error_on_missing_field(key, 'domains', ExitCodes::MISSING_DOMAINS)
+    difficulty = key['difficulty']
+    unless SupportedDifficulties.include?(difficulty)
+        STDERR.puts "Entry '#{key['name']}' has unexpected 'difficulty' field:"\
+                    "'#{difficulty}'.\n\t Use one of #{SupportedDifficulties}"
```

Comment:
Tabbing the suggested line seems strange to me. I'd rather see that on its own line with the suggested values indented.
```suggestion
                    "'#{difficulty}'.\n"\
                    "Use one of the supported difficulty values:\n"\
                    "\t#{SupportedDifficulties}"
```